# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLTemplateElement.h
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 dom/html/HTMLTemplateElement.h
--- a/dom/html/HTMLTemplateElement.h	Wed May 15 03:01:18 2024 +0000
+++ b/dom/html/HTMLTemplateElement.h	Wed May 15 03:01:19 2024 +0000
@@ -65,6 +65,13 @@
     SetHTMLBoolAttr(nsGkAtoms::shadowrootclonable, aValue, aRv);
   }
 
+  bool ShadowRootSerializable() const {
+    return GetBoolAttr(nsGkAtoms::shadowrootserializable);
+  }
+  void SetShadowRootSerializable(bool aValue, ErrorResult& aRv) {
+    SetHTMLBoolAttr(nsGkAtoms::shadowrootserializable, aValue, aRv);
+  }
+
   MOZ_CAN_RUN_SCRIPT
   void SetHTMLUnsafe(const nsAString& aHTML) final;
 